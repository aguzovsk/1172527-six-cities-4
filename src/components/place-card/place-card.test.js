import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCard from './place-card.jsx';
import offersAmsterdam from '../../test-mock/offers-amsterdam.js';

describe(`Test PlaceCard component`, () => {
  const onMouseEnter = () => {};
  const onMouseLeave = () => {};
  const onTitleClick = () => {};

  it(`Test PlaceCard with first hotel, i.e. is premium, not favourite`, () => {
    const tree = renderer.create(
        <PlaceCard
          offer={offersAmsterdam[0]}
          onMouseEnter={onMouseEnter}
          onMouseLeave={onMouseLeave}
          onTitleClick={onTitleClick}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(`Test PlaceCard with second hotel, i.e. not premium, is favourite`, () => {
    const tree = renderer.create(
        <PlaceCard
          offer={offersAmsterdam[1]}
          onMouseEnter={onMouseEnter}
          onMouseLeave={onMouseLeave}
          onTitleClick={onTitleClick}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(`Test PlaceCard with third hotel, i.e. not premium, not favourite`, () => {
    const tree = renderer.create(
        <PlaceCard
          offer={offersAmsterdam[2]}
          onMouseEnter={onMouseEnter}
          onMouseLeave={onMouseLeave}
          onTitleClick={onTitleClick}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
