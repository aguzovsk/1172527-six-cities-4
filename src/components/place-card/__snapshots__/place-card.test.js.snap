// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlaceCard component Test PlaceCard with first hotel, i.e. is premium, not favourite 1`] = `
<article
  className="undefined place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="undefined place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/hotel/3.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €288
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button false"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "60%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Waterfront with extraordinary view
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private room
    </p>
  </div>
</article>
`;

exports[`Test PlaceCard component Test PlaceCard with second hotel, i.e. not premium, is favourite 1`] = `
<article
  className="undefined place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="undefined place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/hotel/17.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €121
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Waterfront with extraordinary view
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private room
    </p>
  </div>
</article>
`;

exports[`Test PlaceCard component Test PlaceCard with third hotel, i.e. not premium, not favourite 1`] = `
<article
  className="undefined place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="undefined place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/hotel/4.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €288
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button false"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "40%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        Amazing and Extremely Central Flat
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private room
    </p>
  </div>
</article>
`;
